cli/gas.c

```diff
@@ -402,8 +402,8 @@ static int gas_read(int argc, char **argv)
 
 	static struct {
 		struct switchtec_dev *dev;
-		unsigned long addr;
-		unsigned long count;
```

Comment:
No, I don't think this is a good idea either. If these stay 'long' instead of 'unsigned long' then you can detect a negative number and do something about it. With 'unsigned long' a negative number becomes a positive.

If you're going to do anything, put an if statement to check for a negative and error out. Or add a CFG_SIZE_SUFFIX or something along those lines.

Reply:
Yes, this patch was modified the previous "unsinged long" to "long", to align with the return value type of handler for CFG_LONG_SUFFIX. After this modified, we can determine whether the number input by user is negative or not. 


Do you feel it is necessary to add two more type/handler pairs:
CFG_POSITIVE_SUFFIX & CFG_NONNEGATIVE_SUFFIX for the arguments here used in gas operations.
gas dump: count (CFG_POSITIVE_SUFFIX)
gas_read: addr (CFG_NONNEGATIVE_SUFFIX) / count (CFG_POSITIVE_SUFFIX)
gas_write: addr (CFG_NONNEGATIVE_SUFFIX)
By these two handler, the addr/count check will be moved to handler, instead of here in gas.c.

Or keep as it is now, the check was in gas.c.

Reply:
That would probably be sufficient. I suggested size_suffix, because this will probably be exclusively used for sizes and you could store the value in a size_t.